src/components/Product.js

```diff
@@ -1,10 +1,10 @@
-const Product = ({ name, description, addToCart }) => {
+export const Product = ({ product, addToCart = () => {}, disableButton }) => {
+  const { name, description } = product
+
   return (
     <>
       <div>{name}: {description}</div>
-      <button>Add to Cart</button>
+      {disableButton ? <></> : <button onClick={() => addToCart(product)}>Add to Cart</button>}
```

Comment:
Shouldn't the `disableButton` prop just disable the button instead of hiding int?

If you wanna hide maybe you can use a short-circuit (https://dev.to/harlessmark/short-circuit-evaluation-with-react-3dn4).

```suggestion
      { !disableButton &&  <button onClick={() => addToCart(product)}>Add to Cart</button> }
```

If you wanna really disable the button you can just pass the `disabled` attribute to the `<button>` element.

Reply:
Done.